### PR TITLE
Enforcer rule to skip non-BOM and non-JAR artifact

### LINKED ISSUE #779: Enforcer rule to skip parent pom
When applied to google-http-java-client's top level pom.xml (`google-http-client-parent`), the enforcer rule fails at the parent pom:

```
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-linkage-checker) @ google-http-client-parent ---
[WARNING] Rule 0: com.google.cloud.tools.dependencies.enforcer.LinkageCheckerRule failed with message:
Artifact com.google.http-client:google-http-client-parent:1.30.2-SNAPSHOT is not associated with a file. The linkage checker enforcer rule should be bound to the verify phase.

```

Because it is a pom artifact, there's no jar file associated with it.

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #779.

### CODE REVIEW

File: enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java [hunk 1]
```diff
@@ -101,9 +102,15 @@ private void setupMock()
         .thenReturn(mockDependencyResolutionResult);
     when(mockRuleHelper.evaluate("${session}")).thenReturn(mockMavenSession);
     when(mockRuleHelper.evaluate("${project}")).thenReturn(mockProject);
+    when(mockProject.getArtifact()).thenReturn(new org.apache.maven.artifact.DefaultArtifact(
+        "com.google.cloud", "linakge-checker-rule-test", "0.0.1", "compile",
```
Comment (elharo) [inline enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java]: linkage

Comment (suztomo) [reply enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java]: Fixed.

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java [hunk 2]
```diff
@@ -116,6 +116,13 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
         logger.warn("The rule is set to read dependency management section but it is empty.");
       }
 
+      String projectType = project.getArtifact().getType();
+      if (dependencySection == DependencySection.DEPENDENCIES && ! "jar".equals(projectType)) {
+        // When checking non-BOM project, not interested in non JAR artifact (parent pom)
```
Comment (elharo) [inline enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: I still don't think you need "parent pom" here. That is not what this is about, or if it is, that's not what we're checking.

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java [hunk 3]
```diff
@@ -116,6 +116,13 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
...
+        logger.info("Skipping project type " + projectType);
```
Comment (elharo) [inline enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: Why log this? Will this appear in Maven output? I doubt users want to see this.

File: enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java (hunk 1 above)
Comment (elharo) [reply enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java]: doesn't look like it is. still spelled linakge

Comment (suztomo) [reply enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java]: (My bad, I fixed another place) Updated now.

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java (hunk 2 above)
Comment (suztomo) [reply enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: Removed "parent pom".

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java (hunk 3 above)
Comment (suztomo) [reply enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: Changed to debug.

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java [hunk 4]
```diff
@@ -116,6 +116,13 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
         logger.warn("The rule is set to read dependency management section but it is empty.");
       }
 
+      String projectType = project.getArtifact().getType();
+      if (dependencySection == DependencySection.DEPENDENCIES && !"jar".equals(projectType)) {
+        // When checking non-BOM project, not interested in non JAR artifact
+        logger.debug("Skipping project type " + projectType);
```
Comment (elharo) [inline enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: Why even log as debug? Who's going to want to read this? Maven outputs far too much junk already.

Comment (suztomo) [reply enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: Removed.

**Comment by suztomo** [comment]
A BOM has pom package type. The enforcer rule should work for a BOM when dependencySection == DependencySection.MANAGED_DEPENDENCIES .

**Pull request merged**